### DISCUSSION TITLE
Setting a cookie for a different domain.

### DISCUSSION THREAD

**Discussion by peter-nitropack** [webdriverio/webdriverio#12096] [Ideas]
### Feature Request
Make it possible to set a cookie for an arbitrary domain.
It makes no sense for such a basic operation to be forbidden and it is required for a great multitude  testing scenarios.

It seems like an arbitrary limitation since it is fundamental yet there is no explanation provided in the [docs](https://webdriver.io/docs/api/browser/setCookies/) for example. If there is a good reason for it, please include it in all relevant places in the documentation.

**Comment by christian-bromann** [comment]
> Make it possible to set a cookie for an arbitrary domain.

This is and probably will never be possible. This would allow me to write automated scripts and hack the internet. There are reasons why such an operation is not allowed. At the end of the day it is also not WebdriverIO who decides whether to support this, it is up to the browser vendor.

**Reply by BorisOsipov** [reply]
>This is and probably will never be possible. This would allow me to write automated scripts and hack the internet. There are reasons why such an operation is not allowed.

but Playwright do it :)

https://playwright.dev/docs/api/class-browsercontext#browser-context-add-cookies

Internally, they spoof a network request to a specific domain (via CDP or similar), then open the spoofed domain in the browser and add cookies & remove the spoof.

**Reply by christian-bromann** [reply]
WebDriver Bidi is introducing browsing contexts too and might allow similar features. Can anyone elaborate on examples where setting cookies for different domains is required?

**Reply by peter-nitropack** [reply]
> [Setting cookies] would allow me to write automated scripts and hack the internet

If only it were that easy :)

**Reply by peter-nitropack** [reply]
Also, the current meta is to visit a page, set your cookies and then you have them the way you want anyway. So it isn't impossible to set the cookies, it is just very, very inconvenient and if you are new to the matter you have to figure it out on your own.

**Comment by peter-nitropack** [comment]
> WebDriver Bidi is introducing browsing contexts too and might allow similar features. Can anyone elaborate on examples where setting cookies for different domains is required?

I am running tests (wdio + mocha) and I have multiple services with different hostnames. Here are some similar examples:

1. You want to test a navigation from one domain to another.
2. You want to test the loading of a resource (e.g. `<script src=...">`) based on a cookie.
